config.go

```diff
@@ -19,6 +20,13 @@ const (
 	DefaultSkipBloomFilters     = false
 )
 
+type MetadataReaders interface {
+	HeaderReader() io.ReaderAt
+	FooterReader() io.ReaderAt
+	PageIndexReader() io.ReaderAt
+	BloomFilterReader() io.ReaderAt
+}
```

Comment:
How about not exporting this interface an simply documenting it?

Reply:
Hmmm, we could keep it internal yes. 

Q: What parameters would the `fileOption` take in that case? `interface{}`?